gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/PubSub.java

```diff
@@ -0,0 +1,194 @@
+
+  Page<Topic> listTopics(ListOption... options);
+
+  Future<AsyncPage<Topic>> listTopicsAsync(ListOption... options);
+
+  String publish(String topic, Message message);
+
+  Future<String> publishAsync(String topic, Message message);
+
+  List<String> publish(String topic, Message message, Message... messages);
+
+  Future<List<String>> publishAsync(String topic, Message message, Message... messages);
+
+  List<String> publish(String topic, Iterable<Message> messages);
+
+  Future<List<String>> publishAsync(String topic, Iterable<Message> messages);
+
+  Subscription create(SubscriptionInfo subscription);
+
+  Future<Subscription> createAsync(SubscriptionInfo subscription);
+
+  // null if not found
+  Subscription getSubscription(String subscription);
+
+  Future<Subscription> getSubscriptionAsync(String subscription);
+
+  void replacePushConfig(String subscription, PushConfig pushConfig);
+
+  Future<Void> replacePushConfigAsync(String subscription, PushConfig pushConfig);
+
+  // false if not found
+  boolean deleteSubscription(String subscription);
+
+  Future<Boolean> deleteSubscriptionAsync(String subscription);
+
+  Page<Subscription> listSubscriptions(ListOption... options);
+
+  Future<AsyncPage<Subscription>> listSubscriptionsAsync(ListOption... options);
+
+  Page<Subscription> listSubscriptions(String topic, ListOption... options);
+
+  Future<AsyncPage<Subscription>> listSubscriptionsAsync(String topic, ListOption... options);
+
+  // Possible Ack options:
+  // 1) return a "special" iterator with "ack-iterated-messages"
+  // 2) provide a way to pull with callback(Message) - ask messages that were called successfully
+  //
+  // Also, consider auto-renewable for all messages that were pulled and not acked.
+  List<ReceivedMessage> pull(String subscription, PullOption... options);
+
+  Future<List<Message>> pullAsync(String subscription, PullOption... options);
+
+  void acknowledge(String subscription, String ackId, String... ackIds);
+
+  Future<Void> acknowledgeAsync(String subscription, String ackId, String... ackIds);
+
+  void acknowledge(String subscription, Iterable<String> ackIds);
+
+  Future<Void> acknowledgeAsync(String subscription, Iterable<String> ackIds);
+
+  void modifyAckDeadline(String subscription, int deadline, TimeUnit unit, String ackId,
+      String... ackIds);
+
+  Future<Void> modifyAckDeadlineAsync(String subscription, int deadline, TimeUnit unit,
+      String ackId, String... ackIds);
+
+  void modifyAckDeadline(String subscription, int deadline, TimeUnit unit, Iterable<String> ackIds);
+
+  Future<Void> modifyAckDeadlineAsync(String subscription, int deadline, TimeUnit unit,
+      Iterable<String> ackIds);
+
+  // Note regarding Data types:
+  // 1) No field selection
+  //  --- This is why there are no options for getters
+  // 2) Never null for primitive values or collections
+  //  --- should we replace default "" with null (e.g. id when not populated)?
```

Comment:
I vote no on replacing the default "" with null.  It could cause some confusion if a user, not knowing proto3, uses the empty string for some special purpose in a message, but then if they check the contents of that message, it's been turned to null.  Also making strings never null would mean less `if` statements in user code.